File: 7_kyu/Sum_of_a_sequence.py
# Returns sum of numbers begin_number,begin_number+step,...,end_number.
def sequence_sum(begin_number: int, end_number: int, step: int) -> int:
    if not (0 <= begin_number <= end_number and 0 < step):
        # Error exit
        return 0
    # Number of summands
    n = 1 + (end_number-begin_number)//step
    # with gauss sum
    return begin_number*n + step*n*(n-1)//2

File: 7_kyu/Sum_of_a_sequence.rs
// Returns sum of start, start+step, start+2*step, ..., end.
fn sequence_sum(start: u32, end: u32, step: u32) -> u32 {
    if end < start {
        // no numbers to add
        return 0;
    };
    // Gauss summation formula
    let number_steps = (end - start) / step;
    // intermediate product in u64 to avoid overflow
    let num = number_steps as u64;
    let n = ((num * (num + 1)) / 2) as u32;
    n * step + (number_steps + 1) * start
}
